src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -69,66 +39,71 @@ BaseCouplingScheme::BaseCouplingScheme(
-                "Maximum time has to be larger than zero!");
+                "Maximum time has to be larger than zero.");
   PRECICE_CHECK(not((maxTimeWindows != UNDEFINED_TIME_WINDOWS) && (maxTimeWindows < 0)),
-                "Maximum number of time windows has to be larger than zero!");
+                "Maximum number of time windows has to be larger than zero.");
   PRECICE_CHECK(not((timeWindowSize != UNDEFINED_TIME_WINDOW_SIZE) && (timeWindowSize < 0.0)),
-                "Time window size has to be larger than zero!");
+                "Time window size has to be larger than zero.");
   PRECICE_CHECK((_validDigits >= 1) && (_validDigits < 17),
-                "Valid digits of timestep length has to be between 1 and 16!");
+                "Valid digits of time window size has to be between 1 and 16.");  // TODO really time window size?
   PRECICE_CHECK(_firstParticipant != _secondParticipant,
-                "First participant and second participant must have different names! Called from BaseCoupling.");
-  if (dtMethod == constants::FIXED_DT) {
+                "First participant and second participant must have different names. Called from BaseCoupling.");
+  if (dtMethod == constants::FIXED_TIME_WINDOW_SIZE) {
     PRECICE_CHECK(hasTimeWindowSize(),
-                  "Timestep length value has to be given when the fixed timestep length method "
-                      << "is chosen for an implicit coupling scheme!");
+                  "Time window size has to be given when the fixed time window size method "
+                      << "is chosen for an implicit coupling scheme.");  // TODO do we talk about window or time step size?
   }
   if (localParticipant == _firstParticipant) {
     _doesFirstStep = true;
-    if (dtMethod == constants::FIRST_PARTICIPANT_SETS_DT) {
-      _participantSetsDt = true;
-      setTimeWindowSize(UNDEFINED_TIME_WINDOW_SIZE);
+    if (dtMethod == constants::FIRST_PARTICIPANT_SETS_TIME_WINDOW_SIZE) {
+      _participantSetsTimeWindowSize = true;
+      _timeWindowSize = UNDEFINED_TIME_WINDOW_SIZE;
     }
   } else if (localParticipant == _secondParticipant) {
-    if (dtMethod == constants::FIRST_PARTICIPANT_SETS_DT) {
-      _participantReceivesDt = true;
+    if (dtMethod == constants::FIRST_PARTICIPANT_SETS_TIME_WINDOW_SIZE) {
+      _participantReceivesTimeWindowSize = true;
     }
   } else {
     PRECICE_ERROR("Name of local participant \""
                   << localParticipant << "\" does not match any "
-                  << "participant specified for the coupling scheme!");
+                  << "participant specified for the coupling scheme.");
   }
   PRECICE_CHECK((maxIterations > 0) || (maxIterations == -1),
-                "Maximal iteration limit has to be larger than zero!");
+                "Maximal iteration limit has to be larger than zero.");
+
+  if (isExplicitCouplingScheme()) {
+    PRECICE_ASSERT(maxIterations == 1);
```

Comment:
Isn't `maxIterations` initialized to "-1"? Should not be set to "1" in my opinion as an explicit cplscheme has no iterations.

Reply:
I agree. A scheme that is not iterating should (ideally) not have any `maxIterations`, at all. Since, this is not possible in our current hierarchy, `maxIterations` should be set to something that is obviously invalid (i.e. `-1`). I will take care of it.

Reply:
Done c0b4fb8986e4f82a96b7e6c61de7e2ca04dcb9dc